game/scripts/vscripts/items/manta.lua

```diff
@@ -0,0 +1,213 @@
+      teamID                --iTeamNumber
+    )
+
+    image:SetForwardVector(forwardVector)
+
+    image:SetPlayerID(playerID)
+    image:SetControllableByPlayer(playerID, true)
+
+    --Level up the image to the caster's level.
+    local level = caster:GetLevel()
+    for i = 1, level - 1 do
+      image:HeroLevelUp(false)
+    end
+
+    --Set the image's available skill points to 0 and teach it the abilities the caster has.
+    image:SetAbilityPoints(0)
+    for abilityIndex = 0, 15 do
+      local casterAbility = caster:GetAbilityByIndex(abilityIndex)
+      if casterAbility ~= nil then
+        local imageAbility = image:FindAbilityByName(casterAbility:GetAbilityName())
+        if imageAbility ~= nil then
+          imageAbility:SetLevel(casterAbility:GetLevel())
+        end
+      end
+    end
+
+    --Recreate the caster's items for the image.
+    for itemSlot = DOTA_ITEM_SLOT_1, DOTA_ITEM_SLOT_5 do
+      local casterItem = caster:GetItemInSlot(itemSlot)
+      if casterItem ~= nil then
+        local imageItem = CreateItem(casterItem:GetName(), image, image)
+        image:AddItem(imageItem)
+      end
+    end
+
+    -- Set Status
+    image:SetHealth(caster:GetHealth())
+    image:SetMana(caster:GetMana())
+
+    -- modifier_illusion controls many illusion properties like +Green damage not adding to the unit damage, not being able to cast spells and the team-only blue particle
+    image:AddNewModifier(caster, ability, "modifier_illusion", {
+      duration = ability:GetSpecialValueFor("illusion_duration"),
+      outgoing_damage = image_outgoing_damage,
+      incoming_damage = image_incoming_damage
+    })
+
+    image:MakeIllusion()  --Without MakeIllusion(), the unit counts as a hero, e.g. if it dies to neutrals it says killed by neutrals, it respawns, etc.  Without it, IsIllusion() returns false and IsRealHero() returns true.
+
+    ability.images[imageIndex] = image
+  end
+
+  -- Allow Render of Caster
+  caster:RemoveNoDraw()
+
+  -- Manta Style End
+  caster:EmitSound("DOTA_Item.Manta.End")
+
+end
+
+-- Vector GetImageLocation(Vector origin, Integer blockedIndex, Boolean ignoreBlock, Integer imageIndex, Integer imageCount)
+function GetImageLocation(origin, blockedIndex, ignoreBlock, imageIndex, imageCount)
+  --[[
+  0: Position of Caster
+  ]]
+
+   if imageIndex == 0 then
+     print('using caster position')
+     return origin
+   end
+
+   if imageIndex >= blockedIndex then
+     if not ignoreBlock then
+       print('index is above blocked index')
+       imageIndex = imageIndex + 1
+     end
+   end
+
+   local distance = Vector(360, 360, 0)
+   local theta = (360 / (imageCount - 1)) * imageIndex
+   local cosTheta = math.cos(theta)
+   local sinTheta = math.sin(theta)
+   local position = Vector(0, 0, 0)
+
+   position.x = (origin.x * cosTheta) - (origin.y * sinTheta)
+   position.y = (origin.x * sinTheta) - (origin.y * cosTheta)
+
+   return (origin + position):Normalized()
```

Comment:
Why rotate a vector if you can create a rotated one already? That's polar coordinates 101.